core/src/apps/wallet/sign_tx/addresses.py

```diff
@@ -239,21 +247,23 @@ def validate_purpose_against_script_type(
     """
     Validates purpose against provided input's script type:
     - 44 for spending address (script_type == SPENDADDRESS)
-    - 48 for multisig (script_type == SPENDMULTISIG)
-    - 49 for p2sh-segwit spend (script_type == SPENDP2SHWITNESS)
-    - 84 for native segwit spend (script_type == SPENDWITNESS)
+    - 45, 48 for multisig (script_type == SPENDMULTISIG)
+    - 49 for p2wsh-nested-in-p2sh spend (script_type == SPENDP2SHWITNESS)
+    - 84 for p2wsh native segwit spend (script_type == SPENDWITNESS)
     """
     if purpose == 44 | HARDENED and script_type != InputScriptType.SPENDADDRESS:
         return False
-    if purpose == 48 | HARDENED and script_type != InputScriptType.SPENDMULTISIG:
+    if purpose == 45 | HARDENED and script_type != InputScriptType.SPENDMULTISIG:
         return False
-    if (  # p2wsh-nested-in-p2sh
-        purpose == 49 | HARDENED and script_type != InputScriptType.SPENDP2SHWITNESS
+    if purpose == 48 | HARDENED and script_type not in (
+        InputScriptType.SPENDMULTISIG,
+        InputScriptType.SPENDP2SHWITNESS,
+        InputScriptType.SPENDWITNESS,
```

Comment:
Could you add a comment why we're allowing P2SHWITNESS and WITNESS here?

Reply:
This is because Electrum uses (wrongly) SPENDP2SHWITNESS/SPENDWITNESS instead of SPENDMULTISIG when doing "Show on Device" for Multisig.